agent/consul/leader_test.go

```diff
@@ -32,6 +32,10 @@ func TestLeader_RegisterMember(t *testing.T) {
 
 	testrpc.WaitForLeader(t, s1.RPC, "dc1")
 
+	if !s1.isReadyForConsistentReads() {
```

Comment:
This is a little weird up in this test, I'd move this to `TestLeader_LeftLeader` where you check that this gets set to false already and make sure it gets set to true when the leader is the leader.